test/extensions_test.rb

```diff
@@ -554,6 +555,19 @@ def process parent, reader, attributes
       assert_equal 0, Asciidoctor::Extensions.groups.size
     end
 
+    # NOTE I'm not convinced we want to continue to support this use case
+    test 'can provide extension registry created without any groups as option' do
+      registry = Asciidoctor::Extensions.create
+      registry.tree_processor SampleTreeProcessor
```

Comment:
Ok.
I'm not sure how to label this feature though. I think we should stick to the DSL and prevent users from creating "detached" objects. But on the other side we want to access those objects to be able to perform actions on them (register, unregister, activate...).

I will create one issue per "use case" so we can discuss on each "use case" if we want to support it and how we should implement it.